app/src/main/java/com/example/habitsmasher/UserDatabaseHelper.java

```diff
@@ -81,14 +88,18 @@ public void onComplete(@NonNull Task<DocumentSnapshot> task) {
      * @param following the following collection of the user
      */
     private void setNumberOfFollowing(List<Object> following) {
-        if (following != null && !following.isEmpty()) {
-            if (following.get(0) == "" && !following.isEmpty()) {
-                _numberOfFollowing.setText("0");
+        try {
+            if (following != null && !following.isEmpty()) {
+                if (following.get(0) == "") {
+                    _numberOfFollowing.setText("0");
+                } else {
+                    _numberOfFollowing.setText(String.valueOf(following.size()));
+                }
             } else {
-                _numberOfFollowing.setText(String.valueOf(following.size()));
+                _numberOfFollowing.setText("0");
             }
-        } else {
-            _numberOfFollowing.setText("0");
+        } catch (NullPointerException e) {
+            Log.d(TAG, "TextView for number following not found");
```

Comment:
Nothing changed here except for the fact that this is now wrapped in a `try/catch` block. An NPE can occur if we are currently not on the profile page, so there's nothing to update. In this case, we don't need to do anything, just log a message for debugging.